src/Examples/Shopping.Api/Services/ShoppingService.cs

```diff
@@ -0,0 +1,56 @@
+using System;
+using System.Threading.Tasks;
+using Grpc.Core;
+using MediatR;
+using Microsoft.Extensions.Logging;
+
+namespace Shopping.Api.Services
+{
+    public class ShoppingService : Api.ShoppingService.ShoppingServiceBase
+    {
+        private readonly ILogger<ShoppingService> _logger;
+        private readonly IMediator _mediator;
+
+        public ShoppingService(ILogger<ShoppingService> logger, IMediator mediator)
+        {
+            _logger = logger;
+            _mediator = mediator;
+        }
+
+        public override async Task Subscribe(IAsyncStreamReader<Request> requestStream, IServerStreamWriter<Response> responseStream, ServerCallContext context)
+        {
+            while (await requestStream.MoveNext())
+            {
+                var request = requestStream.Current;
+                CommandAcknowledgement ack;
+
+                switch (request.CommandMessageCase)
+                {
```

Comment:
Was this the stuff you're thinking of auto-genning? I think we could come up with something nice here.

Reply:
Yep. Probably needs a bit of thought, but a lot of how it works right now is pretty repetitive